test/services/licences/fetch-agreements.service.test.js

```diff
@@ -30,24 +30,42 @@ describe('Fetch Agreements service', () => {
         code: 'S127'
       })
 
-      await LicenceAgreementHelper.add({
+      licenceAgreementData = {
         endDate: new Date('2040-05-01'),
         financialAgreementId: financialAgreement.id,
         licenceRef,
         startDate: new Date('2022-04-01'),
         signedOn: new Date('2022-04-01')
+      }
+    })
+
+    describe('and the agreement has not been deleted', () => {
+      beforeEach(async () => {
+        await LicenceAgreementHelper.add(licenceAgreementData)
+      })
+
+      it('returns the matching agreements data', async () => {
+        const results = await FetchAgreementsService.go(licenceRef)
+
+        expect(results[0]).to.equal({
+          startDate: new Date('2022-04-01'),
+          signedOn: new Date('2022-04-01'),
+          endDate: new Date('2040-05-01'),
+          financialAgreement: { id: '970168ce-06c3-4823-b84d-9da30b742bb8', code: 'S127' }
+        }, { skip: ['id'] })
       })
     })
 
-    it('returns the matching agreements data', async () => {
-      const results = await FetchAgreementsService.go(licenceRef)
+    describe('and the agreement has been deleted', () => {
+      beforeEach(async () => {
+        await LicenceAgreementHelper.add(licenceAgreementData.deletedAt = new Date())
```

Comment:
Sorry, I should have spotted this the first time round. This test is not using your dummy `licenceAgreementData`. What you are actually doing here is using the default helper data and just adding an `endDate` to it. The test is passing because the record you are creating has a different licence reference. I would just put separate `licenceAgreementData` in the `beforeEach` for the "deleted" and " not deleted" tests.

Reply:
Actually I don't think this is even adding an `endDate`

Reply:
Ahh makes sense, thanks 👍🏾